### PR TITLE
📦  FEAT: add react-router + manifest

### PR DESCRIPTION
**Pull request by chancesmith**
## overall purpose
Need to show routing and get script from manifest

## changes made
- add react-router
- export app as SPA + manifest

### CODE REVIEW

File: frontend/index.html [hunk 1]
```diff
@@ -7,7 +7,7 @@
     <title>Vite + React + TS</title>
   </head>
   <body>
-    <div id="root"></div>
+    <div id="new-features"></div>
     <script type="module" src="/src/main.tsx"></script>
```
Comment (chancesmith) [inline frontend/index.html:11]: changing root app id to not match parent app id

File: frontend/src/App.tsx [hunk 2]
```diff
@@ -1,15 +1,38 @@
...
+              <>
+                <p>Sub route /test</p>
+                <br />
+                <Link to={"/"}>Go Back</Link>
+              </>
+            }
+          />
+        </Routes>
+      </div>
+    </BrowserRouter>
```
Comment (chancesmith) [inline frontend/src/App.tsx]: add react-router

File: frontend/src/main.tsx [hunk 3]
```diff
@@ -3,8 +3,10 @@ import ReactDOM from "react-dom/client";
 import App from "./App";
 import "./index.css";
 
-ReactDOM.createRoot(document.getElementById("root") as HTMLElement).render(
+ReactDOM.createRoot(
+  document.getElementById("new-features") as HTMLElement
+).render(
```
Comment (chancesmith) [inline frontend/src/main.tsx:8]: target new root id

File: frontend/vite.config.ts [hunk 4]
```diff
@@ -1,15 +1,19 @@
...
   build: {
-    lib: {
-      entry: "src/App.tsx",
-      name: "AAFeats",
-      formats: ["es"],
-      fileName: "aa-feats",
+    manifest: true,
+    rollupOptions: {
+      input: "src/main.tsx",
     },
```
Comment (chancesmith) [inline frontend/vite.config.ts:17]: change build to deliver a manifest

**Review by nicoledodge — APPROVED** [review]

File: frontend/src/App.tsx [hunk 5]
```diff
@@ -1,15 +1,40 @@
...
-    </div>
+    <BrowserRouter basename="/new-features">
+      <div className="App">
+        Embed App
+        <Routes>
+          <Route
+            path="/"
+            element={
+              <>
+                <h3>Child Application v{VERSION}</h3>
```
Comment (chancesmith) [inline frontend/src/App.tsx:18]: added basename to have relative routing based on the parent app route

**Review by nicoledodge — APPROVED** [review]

**Pull request merged**